src/mon/MDSMonitor.cc

```diff
@@ -2385,16 +2390,14 @@ int MDSMonitor::legacy_filesystem_command(
   if (prefix == "mds set_max_mds") {
     // NOTE: deprecated by "fs set max_mds"
     int64_t maxmds;
-    if (!cmd_getval(g_ceph_context, cmdmap, "maxmds", maxmds) || maxmds < 0) {
+    if (!cmd_getval(g_ceph_context, cmdmap, "maxmds", maxmds) || maxmds <= 0) {
       return -EINVAL;
     }
 
     const MDSMap& mdsmap =
       pending_fsmap.filesystems.at(pending_fsmap.legacy_client_fscid)->mds_map;
       
-    if (!mdsmap.allows_multimds() &&
-	maxmds > mdsmap.get_max_mds() &&
-	maxmds > 1) {
+    if (!mdsmap.allows_multimds() && maxmds > 1) {
```

Comment:
Nope. We need to deal here with upgraded clusters that were set up with multimds prior to the lockout.